catalog/dags/data_refresh/distributed_reindex.py

```diff
@@ -0,0 +1,403 @@
+    launch_template_version_number: int | str,
+    ec2_hook: EC2Hook = None,
+):
+    """
+    Create a new EC2 instance using the launch template for the target
+    environment. In local development, this step is skipped.
+    """
+    if environment != PRODUCTION:
+        raise AirflowSkipException("Skipping instance creation in local environment.")
+
+    instances = ec2_hook.conn.run_instances(
+        MinCount=1,
+        MaxCount=1,
+        LaunchTemplate={
+            "LaunchTemplateName": INDEXER_LAUNCH_TEMPLATES.get(target_environment),
+            "Version": str(launch_template_version_number),
+        },
+        # Name the instance by applying a tag
+        TagSpecifications=[
+            {
+                "ResourceType": "instance",
+                "Tags": [
+                    {
+                        "Key": "Name",
+                        "Value": f"catalog-indexer-worker-{target_environment}",
+                    },
+                ],
+            },
+        ],
+    )["Instances"]
+
+    if not len(instances) == 1:
+        raise Exception(
+            f"Expected one new instance, but {len(instances)} were created."
+        )
+
+    return instances[0]["InstanceId"]
+
+
+@task.sensor(poke_interval=60, timeout=3600, mode="reschedule")
+@setup_ec2_hook
+def wait_for_worker(
+    environment: str,
+    instance_id: str,
+    ec2_hook: EC2Hook = None,
+):
+    """Await the EC2 instance with the given id to be in a healthy running state."""
+    if environment != PRODUCTION:
+        raise AirflowSkipException("Skipping instance creation in local environment.")
+
+    result = ec2_hook.conn.describe_instance_status(InstanceIds=[instance_id])
+
+    instance_status = result.get("InstanceStatuses", [])[0]
+    state = instance_status.get("InstanceState", {}).get("Name")
+    status = next(
+        (
+            status.get("Status")
+            for status in instance_status.get("InstanceStatus", {}).get("Details", [])
+            if status.get("Name") == "reachability"
+        ),
+        None,
+    )
+
+    return PokeReturnValue(
+        # Sensor completes only when the instance is running and has finished initializing
+        is_done=(state == "running" and status == "ok")
+    )
+
+
+@task(
+    # Run locally when create instance tasks are skipped
+    trigger_rule=TriggerRule.NONE_FAILED
+)
+@setup_ec2_hook
+def get_instance_ip_address(
+    environment: str,
+    instance_id: str,
+    ec2_hook: EC2Hook = None,
+):
+    if environment != PRODUCTION:
+        return "catalog_indexer_worker"
+
+    reservations = ec2_hook.describe_instances(instance_ids=[instance_id]).get(
+        "Reservations"
+    )
+    return reservations[0].get("Instances", {})[0].get("PrivateIpAddress")
+
+
+@task(
+    # Instance creation tasks are skipped locally, but we still want this task to run.
+    trigger_rule=TriggerRule.NONE_FAILED
+)
+def create_connection(
+    environment: str,
+    target_environment: str,
```

Comment:
`environment` and `target_environment` are not used in this function

Reply:
Thanks for noting these!